### PR TITLE
Implement boost hash for string_view

### LINKED ISSUE #44: Define boost::hash (and maybe std::hash) for string_view
This is a feature request.

The requirement that it should return the same hash for equal `std::string` should be met.

Additionally, if possible, provide `std::hash<string_view>` with the above requirement, although this is probably not possible to do efficiently because there is no thing like `std::hash(char* ptr, size_t len)`.

### PR DESCRIPTION
**Pull request by dimztimz**
Fixes #44

### CODE REVIEW

File: include/boost/utility/string_view.hpp [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include <boost/detail/workaround.hpp>
 #include <boost/utility/string_view_fwd.hpp>
 #include <boost/throw_exception.hpp>
+#include <boost/container_hash/hash.hpp>
```
Comment (Lastique) [inline include/boost/utility/string_view.hpp]: `hash_fwd.hpp` should be enough here.

**Comment by dimztimz** [comment]
Please merge this.

**Comment by mclow** [comment]
Please remind me to merge this to master after the tests have cycled.

**Comment by dimztimz** [comment]
This is still not merged into master. Travis CI tests seem to pass.

**Pull request merged**